File: examples/java/kora-java-grpc-client/src/main/java/io/koraframework/example/grpc/client/MyClientInterceptor.java
package io.koraframework.example.grpc.client;

import io.grpc.*;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import io.koraframework.common.annotation.Component;
import io.koraframework.common.annotation.Tag;
import io.koraframework.generated.grpc.UserServiceGrpc;

@Tag(UserServiceGrpc.class)
@Component
public final class MyClientInterceptor implements ClientInterceptor {

    private final Logger logger = LoggerFactory.getLogger(MyClientInterceptor.class);

    @Override
    public <ReqT, RespT> ClientCall<ReqT, RespT>
            interceptCall(MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
        logger.info("INTERCEPTED");
        return next.newCall(method, callOptions);
    }
}

File: examples/kotlin/kora-kotlin-grpc-client/src/main/kotlin/io/koraframework/kotlin/example/grpc/client/MyClientInterceptor.kt
package io.koraframework.kotlin.example.grpc.client

import io.grpc.*
import org.slf4j.LoggerFactory
import io.koraframework.common.annotation.Component
import io.koraframework.common.annotation.Tag
import io.koraframework.common.annotation.Root
import io.koraframework.generated.grpc.UserServiceGrpc

@Tag(UserServiceGrpc::class)
@Component
class MyClientInterceptor : ClientInterceptor {
    private val logger = LoggerFactory.getLogger(MyClientInterceptor::class.java)

    override fun <ReqT : Any, RespT : Any> interceptCall(
        method: MethodDescriptor<ReqT, RespT>,
        callOptions: CallOptions,
        next: Channel
    ): ClientCall<ReqT, RespT> {
        logger.info("INTERCEPTED")
        return next.newCall(method, callOptions)
    }
}

